File: packages/php/src/Resources/Health.php
<?php

declare(strict_types=1);

namespace MissionInbox\Resources;

use MissionInbox\Http\Client;

/**
 * The `health` resource. Access via `$mi->health`.
 *
 * Unauthenticated endpoint. Returns the API's health string on success.
 */
final class Health
{
    public function __construct(private readonly Client $http) {}

    /** Ping the API. Returns the health status string or associative array. */
    public function check(): string|array
    {
        $result = $this->http->request('GET', '/api/health');
        return is_array($result) ? $result : (string) $result;
    }
}

File: packages/java/src/main/java/com/missioninbox/resources/Health.java
package com.missioninbox.resources;

import com.missioninbox.http.HttpClient;

/**
 * The {@code health} resource. Access via {@code mi.health}.
 *
 * <p>Unauthenticated endpoint. Returns the API's health string on success.
 */
public final class Health {

    private final HttpClient http;

    public Health(HttpClient http) {
        this.http = http;
    }

    /** Ping the API. */
    public Object check() {
        return http.request("GET", "/api/health", null, null);
    }
}
